homeassistant/components/bryant_evolution/climate.py

```diff
@@ -0,0 +1,246 @@
+
+        raise HomeAssistantError(
+            translation_domain=DOMAIN,
+            translation_key="failed_to_parse_hvac_mode",
+            translation_placeholders={"mode": mode},
+        )
+
+    async def _read_hvac_action(self) -> HVACAction:
+        """Return the current running hvac operation."""
+        mode_and_active = await self._client.read_hvac_mode()
+        if not mode_and_active:
+            raise HomeAssistantError(
+                translation_domain=DOMAIN, translation_key="failed_to_read_hvac_action"
+            )
+        mode, is_active = mode_and_active
+        if not is_active:
+            return HVACAction.OFF
+        match mode.upper():
+            case "HEAT":
+                return HVACAction.HEATING
+            case "COOL":
+                return HVACAction.COOLING
+            case "OFF":
+                return HVACAction.OFF
+            case "AUTO":
+                # In AUTO, we need to figure out what the actual mode is
+                # based on the setpoints.
+                if (
+                    self.current_temperature is not None
+                    and self.target_temperature_low is not None
+                ):
+                    if self.current_temperature > self.target_temperature_low:
+                        # If the system is on and the current temperature is
+                        # higher than the point at which heating would activate,
+                        # then we must be cooling.
+                        return HVACAction.COOLING
+                    return HVACAction.HEATING
+        raise HomeAssistantError(
+            translation_domain=DOMAIN,
+            translation_key="failed_to_parse_hvac_mode",
+            translation_placeholders={
+                "mode_and_active": mode_and_active,
+                "current_temperature": str(self.current_temperature),
+                "target_temperature_low": str(self.target_temperature_low),
+            },
+        )
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new target hvac mode."""
+        if hvac_mode == HVACMode.HEAT_COOL:
+            hvac_mode = HVACMode.AUTO
+        if not await self._client.set_hvac_mode(hvac_mode):
+            raise HomeAssistantError(
+                translation_domain=DOMAIN, translation_key="failed_to_set_hvac_mode"
+            )
+        self._attr_hvac_mode = hvac_mode
+        self._async_write_ha_state()
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        if kwargs.get("target_temp_high"):
+            temp = int(kwargs["target_temp_high"])
+            if not await self._client.set_cooling_setpoint(temp):
+                raise HomeAssistantError(
+                    translation_domain=DOMAIN, translation_key="failed_to_set_clsp"
+                )
+            self._attr_target_temperature_high = temp
+            self._async_write_ha_state()
```

Comment:
I need to double check if this function doesn't result in 3 state writes in one action

Reply:
signal-boost

Reply:
fixed as discussed